osquery/logger/plugins/aws_firehose.cpp

```diff
@@ -105,20 +105,17 @@ Status FirehoseLogForwarder::setUp() {
   }
 
   // Make sure we can connect to designated stream
-  Aws::Firehose::Model::ListDeliveryStreamsRequest r;
-  auto result = client_->ListDeliveryStreams(r).GetResult();
-  std::vector<std::string> stream_names = result.GetDeliveryStreamNames();
-
-  if (std::find(stream_names.begin(),
-                stream_names.end(),
-                FLAGS_aws_firehose_stream) == stream_names.end()) {
-    return Status(1,
-                  "Could not find Firehose delivery stream: " +
-                      FLAGS_aws_firehose_stream);
+  Aws::Firehose::Model::DescribeDeliveryStreamRequest r;
+  r.SetDeliveryStreamName(FLAGS_aws_firehose_stream);
+  auto result = client_->DescribeDeliveryStream(r).GetResult();
+  auto description = result.GetDeliveryStreamDescription();
+  auto stream_name = description.GetDeliveryStreamName();
+  if (stream_name.empty()){
```

Comment:
Based on the typedef for the result of that call, it doesn't look like there should be a concern about segfaults: http://sdk.amazonaws.com/cpp/api/0.11.0-7-g6be9f8d/d9/d8b/namespace_aws_1_1_firehose_1_1_model.html#a5b01d7c60c14aa75db9edd709e98d599

However, it would make the code more clear if we did a call to `IsSuccess()` before `GetResult()`.